File: next/security/backend-ms1/src/main/java/ma/zs/stocky/ws/converter/security/ActionPermissionConverter.java
package  ma.zs.stocky.ws.converter.security;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;




import ma.zs.stocky.zynerator.util.StringUtil;
import ma.zs.stocky.zynerator.converter.AbstractConverter;
import ma.zs.stocky.zynerator.util.DateUtil;
import ma.zs.stocky.bean.core.security.ActionPermission;
import ma.zs.stocky.ws.dto.security.ActionPermissionDto;

@Component
public class ActionPermissionConverter extends AbstractConverter<ActionPermission, ActionPermissionDto> {


    public  ActionPermissionConverter() {
        super(ActionPermission.class, ActionPermissionDto.class);
    }

    @Override
    public ActionPermission toItem(ActionPermissionDto dto) {
        if (dto == null) {
            return null;
        } else {
        ActionPermission item = new ActionPermission();
            if(StringUtil.isNotEmpty(dto.getId()))
                item.setId(dto.getId());
            if(StringUtil.isNotEmpty(dto.getReference()))
                item.setReference(dto.getReference());
            if(StringUtil.isNotEmpty(dto.getLibelle()))
                item.setLibelle(dto.getLibelle());



        return item;
        }
    }

    @Override
    public ActionPermissionDto toDto(ActionPermission item) {
        if (item == null) {
            return null;
        } else {
            ActionPermissionDto dto = new ActionPermissionDto();
            if(StringUtil.isNotEmpty(item.getId()))
                dto.setId(item.getId());
            if(StringUtil.isNotEmpty(item.getReference()))
                dto.setReference(item.getReference());
            if(StringUtil.isNotEmpty(item.getLibelle()))
                dto.setLibelle(item.getLibelle());


        return dto;
        }
    }




    public void initObject(boolean value) {
    }


}

File: nest-angular/security-v2/backend-ms1/src/app/shared/converter/security/ActionPermissionConverter.ts
import {forwardRef, Inject, Injectable} from "@nestjs/common";
import {AbstractConverter} from "src/app/zynerator/converter/AbstractConverter";

import {ActionPermission} from "src/app/shared/bean/core/security/ActionPermission";
import {ActionPermissionDto} from "src/app/shared/dto/security/ActionPermissionDto";


@Injectable()
export class ActionPermissionConverter extends AbstractConverter<ActionPermission, ActionPermissionDto> {


    constructor(
        ) {
            super();
            this.init(true);
    }

    toItem(dto: ActionPermissionDto): ActionPermission {
        if (!dto) {
            return null;
        }
        const item =new ActionPermission();
        if (dto.id) {
            item.id = dto.id;
        }
        if (dto.reference) {
            item.reference = dto.reference;
        }
        if (dto.libelle) {
            item.libelle = dto.libelle;
        }




        return item;
    }

    toDto(item: ActionPermission): ActionPermissionDto {
        if (!item) {
            return null;
        }
        const dto = new ActionPermissionDto();

        if (item.id) {
            dto.id = item.id;
        }
        if (item.reference) {
            dto.reference = item.reference;
        }
        if (item.libelle) {
            dto.libelle = item.libelle;
        }


        return dto;
    }



    public initObject(value: boolean): void {
    }
}
